### PR TITLE
Feature/adding APs to cisco wlc

### PR DESCRIPTION
**Pull request by mogensen**
Adding APs to Cisco WLC on discovery.
Polling active users on APs.

### CODE REVIEW

**Comment by laf** [comment]
Does this change anything for existing setups other than just new / more information?

**Comment by fmotrifork** [comment]
This only adds more info from the Cisco wlc. Using the same strategy as in the aruba controller.

**Comment by fmotrifork** [comment]
It results in a list looking like this:

image

**Comment by mogensen** [comment]
This only adds more info from the Cisco Wireless Controller. Using the same strategy as in the aruba controller.

It results in a list looking like this:
image

File: includes/polling/os/ciscowlc.inc.php [hunk 1]
```diff
@@ -75,8 +64,122 @@
...
+
+foreach ($radios as $key => $value) {
+ 
+    $indexName = substr($key, 0, -2);
+
+    $channel       = str_replace('ch', '', $value['bsnAPIfPhyChannelNumber']);
+    $mac           = str_replace(' ', ':', $stats[$indexName]['bsnAPDot3MacAddress']);
+    $name          = $stats[$indexName]['bsnAPName'];
+    $numasoclients = $value['bsnApIfNoOfUsers'];
+    $radionum      = array_pop(explode('.',$key));
```
Comment (laf) [inline includes/polling/os/ciscowlc.inc.php]: Can you take a look at this please.

**Comment by laf** [comment]
One in line comment from scurtinizer run.

File: includes/polling/os/ciscowlc.inc.php (hunk 1 above)
Comment (mogensen) [inline includes/polling/os/ciscowlc.inc.php]: Could you specify a bit more what you want me to look at?

Comment (laf) [inline includes/polling/os/ciscowlc.inc.php]: Sorry I've only just realised I see the scrut comments from a plugin :/

`explode('.', $key) cannot be passed to array_pop() as the parameter $array expects a reference.`

**Comment by PENorris** [comment]
Cheers, for adding this. This keeps me from missing Prime. 
Is there any way to keep APs that disassociated in the list and mark them in some way as no longer associated?

**Comment by mogensen** [comment]
@Penorris i have been playing with that plan as well. ;) It may happen some day. ;)

**Comment by FTBZ** [comment]
Many problem with this implementation, it's using many things from the Aruba code and it's making a mix between Cisco WLC and Aruba. Cisco AP name is arubaXXX and the graph is not from the Cisco WLC but from the Aruba wifi_client? It's really confusing to work on.

**Comment by laf** [comment]
I've locked this. You should create an issue if there is a problem to be fixed or if you want to discuss things then you can do so on community.librenms.org

**Pull request merged**